feat#3-Add chromosome methylation map prep and Holoviews visuals

Michael14567 opened the pull request:
Add chrmap data prep (line/box) with smoothing and per-chromosome ticks/borders Add Holoviews helpers for line and box/violin plots; generate sample HTML assets from test data Export new APIs via bsx2.plots and cover with smoke tests

<img width="474" height="404" alt="изображение" src="https://github.com/user-attachments/assets/39c00601-3be6-4a8f-befe-5d21c77b89dd" />
<img width="459" height="405" alt="изображение" src="https://github.com/user-attachments/assets/d601e974-a761-44c2-8ee6-e98a0f9b7ef1" />

Michael14567 committed:
feat#3-Add chromosome methylation map prep and Holoviews visuals

Add chrmap data prep (line/box) with smoothing and per-chromosome ticks/borders
Add Holoviews helpers for line and box/violin plots; generate sample HTML assets from test data
Export new APIs via bsx2.plots and cover with smoke tests

shitohana commented on python/src/bsx2/plots/chrmap.py:
```diff
@@ -0,0 +1,142 @@
+    # adjust length if even window
+    if smoothed.shape[0] > values.shape[0]:
+        smoothed = smoothed[: values.shape[0]]
+    return smoothed
```
Скользящее среднее отстает при сглаживании. Первая версия bsxplorer использовала https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.savgol_filter.html. Дает не отстающие, адекватные результаты

Michael14567 committed:
Use Savitzky-Golay smoothing for chrmap lines

replace moving-average smoothing in prepare_chr_line_data with scipy.signal.savgol_filter
